src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsService.java

```diff
@@ -119,6 +126,45 @@ public void updatePlanRepository(String buildProjectKey, String buildPlanKey, St
         jenkinsBuildPlanService.updateBuildPlanRepositories(buildProjectKey, buildPlanKey, newRepoUrl, existingRepoUrl);
     }
 
+    @Override
+    public List<Long> getAllArtemisBuildPlanServerNotificationIds(String buildPlanKey) {
+        return null;
+    }
+
+    @Override
+    public void deleteBuildPlanServerNotificationId(String buildPlanKey, Long serverNotificationId) {
+        throw new RuntimeException("Not implemented");
+    }
+
+    @Override
+    public void createBuildPlanServerNotification(String buildPlanKey, String serverNotificationUrl) {
+        throw new RuntimeException("Not implemented");
+    }
+
+    @Override
+    public void fixBuildPlanNotification(String projectKey, String buildPlanKey, VcsRepositoryUrl repositoryUrl) {
+        String jobName = buildPlanKey.replace(projectKey + "-", "");
+        Document config;
+        try {
+            config = jenkinsJobService.getJobConfig(projectKey, jobName);
+        }
+        catch (IOException e) {
+            log.error("Could not fix build plan notification for build plan " + buildPlanKey + " in project " + projectKey, e);
+            return;
+        }
+        String stringConfig = config.getTextContent();
+        Pattern pattern = Pattern.compile("(.*?notificationUrl: ')(.+?)('.*?)");
+        Matcher matcher = pattern.matcher(stringConfig);
+        String newStringConfig = matcher.replaceAll("$1" + artemisServerUrl + NEW_RESULT_RESOURCE_API_PATH + "$3");
+        config.setTextContent(newStringConfig);
```

Comment:
Could be moved into a separate small private helper method, then this method would follow a very clear sequence
- fetch current config
- `updateBuildPlanNotificationUrl(config, …)`
- push new config back to Jenkins.